File: frontend/src/api/skill.ts
import { request } from '@/utils/http-client'

export interface SkillData {
  id?: number
  name: string
  description: string
  toolNames: string
  instructions: string
  triggerWords?: string
  confidence?: number
  usageCount?: number
  successCount?: number
  failCount?: number
  userId: number
  agentType?: string
  agentConfigId?: number
  createdAt?: string
  updatedAt?: string
}

export async function listSkills(userId: number, agentConfigId?: number) {
  const params = new URLSearchParams({ userId: String(userId) })
  if (agentConfigId) params.set('agentConfigId', String(agentConfigId))
  return request<SkillData[]>(`/skills?${params}`)
}

export async function createSkill(data: Partial<SkillData>) {
  return request<{ success: boolean; data: SkillData; message: string }>('/skills', {
    method: 'POST',
    body: JSON.stringify(data)
  })
}

export async function updateSkill(id: number, data: Partial<SkillData>) {
  return request<{ success: boolean; data: SkillData; message: string }>(`/skills/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data)
  })
}

export async function deleteSkillApi(id: number, userId: number) {
  return request<{ success: boolean; message: string }>(`/skills/${id}?userId=${userId}`, { method: 'DELETE' })
}

File: src/main/java/com/example/agentdeepseek/model/entity/Skill.java
package com.example.agentdeepseek.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 技能实体类
 * AI 自动管理的可复用工作流单元
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Skill {
    private Long id;
    private String name;
    private String description;
    private String toolNames;
    private String instructions;
    private String triggerWords;
    private Double confidence;
    private Integer usageCount;
    private Integer successCount;
    private Integer failCount;
    private Long userId;
    private String agentType;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Skill(String name, String description, String toolNames, String instructions,
                 Long userId, String agentType) {
        this.name = name;
        this.description = description;
        this.toolNames = toolNames;
        this.instructions = instructions;
        this.confidence = 0.5;
        this.usageCount = 0;
        this.successCount = 0;
        this.failCount = 0;
        this.userId = userId;
        this.agentType = agentType;
    }

    public Skill(String name, String description, String toolNames, String instructions,
                 String triggerWords, Long userId, String agentType) {
        this(name, description, toolNames, instructions, userId, agentType);
        this.triggerWords = triggerWords;
    }
}
